tests/CMakeLists.txt

```diff
@@ -106,35 +106,28 @@ ENDIF()
 set(AlbanyPath                         ${Albany_BINARY_DIR}/src/Albany)
 set(AlbanyAnalysisPath                 ${Albany_BINARY_DIR}/src/AlbanyAnalysis)
 
-IF (ALBANY_MPI)
-  set(SerialAlbany.exe                 ${SERIAL_CALL} ${AlbanyPath})
-  set(SerialAlbanyAnalysis.exe         ${SERIAL_CALL} ${AlbanyAnalysisPath})
-  # Do not test on greater than Trilinos_MPI_EXEC_MAX_NUMPROCS configured in Trilinos build
-  # or explicitly given at Albany configure time -D ALBANY_MPI_EXEC_MAX_NUMPROCS
-  IF(DEFINED MPIMNP)
-    IF(MPIMNP LESS 8)
-      SET(MAX_MPI_RANKS ${MPIMNP})
-    ELSE(MPIMNP LESS 8)
-      SET(MAX_MPI_RANKS 8)
-    ENDIF(MPIMNP LESS 8)
-  ELSE(DEFINED MPIMNP)
-      SET(MAX_MPI_RANKS 4)
-  ENDIF(DEFINED MPIMNP)
-  IF (ALBANY_ENABLE_CUDA) 
-    set(Albany.exe                     ${PARALLEL_CALL} ${AlbanyPath} ${KOKKOS_NDEVICES})
-    set(Albany8.exe                    ${MPIEX} ${MPIPRE} ${MPINPF} ${MAX_MPI_RANKS} ${MPIPOST} ${AlbanyPath} ${KOKKOS_NDEVICES})
-    set(AlbanyAnalysis.exe             ${PARALLEL_CALL} ${AlbanyAnalysisPath} ${KOKKOS_NDEVICES})
-  ELSE() 
-    set(Albany.exe                     ${PARALLEL_CALL} ${AlbanyPath})
-    set(Albany8.exe                    ${MPIEX} ${MPIPRE} ${MPINPF} ${MAX_MPI_RANKS} ${MPIPOST} ${AlbanyPath})
-    set(AlbanyAnalysis.exe             ${PARALLEL_CALL} ${AlbanyAnalysisPath})
-  ENDIF() 
-ELSE()
-  set(SerialAlbany.exe                 ${AlbanyPath})
-  set(SerialAlbanyAnalysis.exe         ${AlbanyAnalysisPath})
-  set(Albany.exe                       ${AlbanyPath})
-  set(AlbanyAnalysis.exe               ${AlbanyAnalysisPath})
-ENDIF()
+set(SerialAlbany.exe                 ${SERIAL_CALL} ${AlbanyPath})
+set(SerialAlbanyAnalysis.exe         ${SERIAL_CALL} ${AlbanyAnalysisPath})
+# Do not test on greater than Trilinos_MPI_EXEC_MAX_NUMPROCS configured in Trilinos build
+# or explicitly given at Albany configure time -D ALBANY_MPI_EXEC_MAX_NUMPROCS
+IF(DEFINED MPIMNP)
+  IF(MPIMNP LESS 8)
+    SET(MAX_MPI_RANKS ${MPIMNP})
+  ELSE(MPIMNP LESS 8)
+    SET(MAX_MPI_RANKS 8)
+  ENDIF(MPIMNP LESS 8)
+ELSE(DEFINED MPIMNP)
+    SET(MAX_MPI_RANKS 4)
+ENDIF(DEFINED MPIMNP)
+IF (ALBANY_ENABLE_CUDA) 
+  set(Albany.exe                     ${PARALLEL_CALL} ${AlbanyPath} ${KOKKOS_NDEVICES})
+  set(Albany8.exe                    ${MPIEX} ${MPIPRE} ${MPINPF} ${MAX_MPI_RANKS} ${MPIPOST} ${AlbanyPath} ${KOKKOS_NDEVICES})
```

Comment:
@mperego I see that Albany8 is only used for AIS tests and the 20km GIS test. What's the reason for this peculiarity? Why can't all test use the same number of ranks for the parallel call?